compiler/rustc_typeck/src/check/op.rs

```diff
@@ -549,16 +549,11 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         is_assign: IsAssign,
         op: hir::BinOp,
     ) -> bool {
-        let source_map = self.tcx.sess.source_map();
-        let remove_borrow_msg = "String concatenation appends the string on the right to the \
-                                 string on the left and may require reallocation. This \
-                                 requires ownership of the string on the left";
-
-        let msg = "`to_owned()` can be used to create an owned `String` \
-                   from a string reference. String concatenation \
-                   appends the string on the right to the string \
-                   on the left and may require reallocation. This \
-                   requires ownership of the string on the left";
+        let str_concat_note = "String concatenation appends the string on the right to the
+string on the left and may require reallocation.
+This requires ownership of the string on the left.";
```

Comment:
```suggestion
this requires ownership of the string on the left.";
```

nit: same as above

Reply:
I agree that the capital letters look a bit weird, but to me, it also looks weird to have multiple sentences that are not capitalized.

Maybe changing it in some way similar to https://github.com/rust-lang/rust/pull/92843#discussion_r785389097 could solve this problem.

Reply:
You could also split in into multiple help messages if that would look better.